src/qibo/gates/special.py

```diff
@@ -54,7 +55,7 @@ def __init__(self, *q):
     def from_gate(cls, gate):
         fgate = cls(*gate.qubits)
         fgate.append(gate)
-        if isinstance(gate, (M, SpecialGate)):
+        if isinstance(gate, (M, SpecialGate, I)):
```

Comment:
@AlejandroSopena could you comment why you need to add `I` here?

Reply:
This was necessary in an initial version of the code, but not anymore. I have removed it.